app/services/report_template_format.rb

```diff
@@ -0,0 +1,45 @@
+class ReportTemplateFormat
+  def self.all
+    [
+      new(id: :csv, mime_type: 'text/csv', file_suffix: 'csv', human_name: 'CSV'),
+      new(id: :txt, mime_type: 'text/plain', file_suffix: 'txt', human_name: 'Plain text'),
+      new(id: :json, mime_type: 'application/json', file_suffix: 'json', human_name: 'JSON'),
+      new(id: :yaml, mime_type: 'text/yaml', file_suffix: 'yaml', human_name: 'YAML'),
+      new(id: :html, mime_type: 'text/html', file_suffix: 'html', human_name: 'HTML')
+    ]
+  end
+
+  def self.selectable
+    all.reject { |f| f.id == :txt }
+  end
+
+  def self.find(id)
+    all.find { |f| f.id.to_s == id.to_s }
+  end
+
+  # if the template does not support formats at all,
+  # most likely custom template
+  # we fallback to universal plaintext format
+  def self.system
+    find(:txt)
+  end
+
+  # if the template supports format selection,
+  # csv is the default
+  def self.default
+    find(:csv)
+  end
+
+  attr_reader :id, :mime_type, :file_suffix, :human_name
+
+  def initialize(id:, mime_type:, file_suffix:, human_name:)
+    @id = id
+    @mime_type = mime_type
+    @file_suffix = file_suffix
+    @human_name = human_name
+  end
+
+  def suffix
+    ".#{@file_suffix}"
```

Comment:
could we name it extension and use the Rails mime type for this, instead of defining it?
`Mime::Type.lookup('image/jpeg').symbol.to_s`

Reply:
tested it worked for every type, implemented and renamed the method :+1: